support/vm/vector_gen.go

```diff
@@ -0,0 +1,407 @@
+package vm
```

Comment:
I think we should loosen the coupling with the VM and move this to its own `support` package. You'll have to widen the public interface to the VM slightly, but I think that's fine. We should be able to use this in principle with a different VM.

Reply:
I'm putting this off until the future where specs-actors takes full responsibility for the format: #1455.  Until then it is not intended for these vector types to be used externally so I will move them to internal types.